src/frontends/common/include/openvino/frontend/extension/conversion.hpp

```diff
@@ -0,0 +1,77 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "openvino/core/extension.hpp"
+#include "openvino/frontend/node_context.hpp"
+#include "openvino/frontend/visibility.hpp"
+
+namespace ov {
+namespace frontend {
+
+class FRONTEND_API ConversionExtensionBase : public ov::Extension {
+public:
+    using Ptr = std::shared_ptr<ConversionExtensionBase>;
+    explicit ConversionExtensionBase(const std::string& op_type) : m_op_type(op_type) {}
+
+    const std::string& get_op_type() const {
+        return m_op_type;
+    }
+
+    ~ConversionExtensionBase() override = 0;
+
+private:
+    std::string m_op_type;
+};
+
+class FRONTEND_API ConversionExtension : public ConversionExtensionBase {
+public:
+    using Ptr = std::shared_ptr<ConversionExtension>;
+    ConversionExtension(const std::string& op_type, const CreatorFunction& converter)
+        : ConversionExtensionBase(op_type),
+          m_converter(converter) {}
+
+    ConversionExtension(const std::string& op_type, const CreatorFunctionNamed& converter)
+        : ConversionExtensionBase(op_type),
+          m_converter_named(converter) {}
+
+    const CreatorFunction& get_converter() {
+        return m_converter;
+    };
+
+    const CreatorFunctionNamed& get_converter_named() {
+        return m_converter_named;
+    };
+
+    ~ConversionExtension() override = default;
+
+    using PyCreatorFunction = std::function<OutputVector(const NodeContext*)>;
```

Comment:
Can you change the function's argument to `const NodeContext&` here? Below you're assigning a lambda which takes a const reference to the `m_converter` member which uses this alias as a typedef. Does it even compile btw?

Reply:
Unfortunately, I can't, it's a workaround to avoid an issue in pybind, I will add a detailed comment about it

Reply:
![image](https://user-images.githubusercontent.com/21181928/148838069-9cafffc7-1307-49c7-b354-a9e78255642d.png)

I tried to explain in detail, if something is not clear, then we can discuss it offline.
Maybe @akuporos know something about this issue

Reply:
I have localized a workaround in pybind wrappers for ConversionExtensions, public classes are not affected now.